Script to pull seqr pedigrees using API

mike-w-wilson opened the pull request:
This script takes in the users email that is used for seqr login as well as a list of seqr project_guids and builds pedigrees from seqr's API.

bw2 commented on pull_seqr_pedigrees.py:
```diff
@@ -0,0 +1,57 @@
+                'password': getpass.getpass(),
+            }
+        )
+        final_ped = open('seqr_pedigrees.txt', 'w')
```
style comment - for consistency, you might want to use the 
with open("a.txt") as a, \
         open("b.txt") as b, ...:
syntax (https://legacy.python.org/dev/peps/pep-0008/#maximum-line-length)

mike-w-wilson replied:
This  is a better approach, thanks!

bw2 commented on pull_seqr_pedigrees.py:
```diff
@@ -0,0 +1,57 @@
+            }
+        )
+        final_ped = open('seqr_pedigrees.txt', 'w')
+        final_ped.write('Project_GUID\tFamily_ID\tIndividual_ID\tPaternal_ID\tMaternal_ID\tSex\n') 
```
also a style thing - but '\t'.join(['column1', 'column2' ... ]) + "\n" reduces chance of typos

mike-w-wilson replied:
I've converted the variable writing to this method but have to cast the paternal and maternal IDs as they are occasionally `None`. I'm not sure which I like more but went with your suggestion here.

bw2 commented on pull_seqr_pedigrees.py:
```diff
@@ -0,0 +1,57 @@
+                else:
+                    fams = r.json()['familiesByGuid']
+                    fams_dict = {}
+                    for key in fams:
```
both here and in the loop below, since key isn't used except to do fams[key], would instead do 
for fam in fams.values()

mike-w-wilson replied:
Much better!

mike-w-wilson commented:
thanks for the input @bw2 ! I've addressed your feedback, formatted with black, and also changed the function a bit to include a main() which is probably why github dismissed your other review. Would you mind taking one more quick look? Thanks!

bw2 commented on pull_seqr_pedigrees.py:
```diff
@@ -0,0 +1,92 @@
+                    inds = r.json()["individualsByGuid"]
+                    for ind in inds.values():
+                        family_id = fams_dict[ind["familyGuid"]]
+                        project_guid = ind["projectGuid"]
+                        individual_id = ind["individualId"]
+                        paternal_id = ind["paternalId"]
+                        maternal_id = ind["maternalId"]
+                        sex = ind["sex"]
```
since these variables are only used once and their names aren't adding to readability, I'd just do
"\t".join([
                                    ind["projectGuid"],
                                    family_id,
                                    ind["individualId"],
                                    ind["paternalId"] or "",
                                    ind["maternalId"] or "",
                                    ind["sex"],
                                ])
Another shortcut if you prefer None to "" is 
```
"\t".join(map(str, [  ind["projectGuid"],  ... ] ))
```

mike-w-wilson replied:
I like that second approach for keeping `None`. @klaricch , do you prefer "None" or "" when a paternal or maternal ID is missing?

klaricch replied:
i think None is preferable

klaricch commented on pull_seqr_pedigrees.py:
```diff
@@ -0,0 +1,92 @@
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--email", help="Email used for seqr login", required=True)
```
can -e also be added here?

klaricch commented on pull_seqr_pedigrees.py:
```diff
@@ -0,0 +1,92 @@
+                        individual_id = ind["individualId"]
+                        paternal_id = ind["paternalId"]
+                        maternal_id = ind["maternalId"]
+                        sex = ind["sex"]
```
maybe for consistency a "U" for sex should also just be converted to None?

mike-w-wilson commented:
@bw2 @klaricch , sorry would you guys mind reviewing one more time? Github dismissed the approval again. Thanks!

bw2 commented:
LGTM!
